### DISCUSSION TITLE
How do I pass parameter to view?

### DISCUSSION THREAD

**Discussion by amitava82** [hasura/graphql-engine#9903] [Question] [answered]
I have a view that does custom aggregation on a table. However, I need to do the aggregation on a particular date range. How do I pass the start and end date parameter dynamically when doing graphql query? For example:
```
CREATE
OR REPLACE VIEW "public"."v_metrics" AS
count(*) AS total
from events
WHERE
    start_at >= 'start_date' AND start_at < 'end_date' 
GROUP BY
  user_id
```
And then from graphql query I should be able to do something like.
```
query {
  v_metrics(where: {start_at: {_eq: "2022-01-01"}, start_at: {_eq: "2022-02-01"}})
}
```

**Comment by SamirTalwar** [comment, marked as answer]
It depends how tied you are to the view. Views don't take parameters; you'll want a function for that.

If you don't really need the view, you could write the query directly; it'd look something like this:

```graphql
query {
  events_aggregate(
    where: {
      _and: [
        {start_at: {_gte: "2022-01-01"}},
        {start_at: {_lt: "2022-02-01"}}
      ]
    }
  ) {
    aggregate {
      count
    }
  }
}
```

This uses the [comparison operators](https://hasura.io/docs/latest/queries/postgres/filters/comparison-operators/) `_gte` for `>=` and `_lt` for `<`.

If you really this to be encoded in the database itself, I recommend using a function, and then [calling that function from Hasura](https://hasura.io/docs/latest/schema/postgres/custom-functions/), or using a [native query](https://hasura.io/docs/latest/schema/postgres/native-queries/).

**Comment by amitava82** [comment]
Understood, yes I do need the view since the query is complex that cannot be done with direct query (above example was a simplified). I'll look at the docs and figure out alternative. At least I know view cannot take parameters.
